### DISCUSSION TITLE
Fetch cache for a particular user

### DISCUSSION THREAD

**Discussion by Amritpal2001** [zilliztech/GPTCache#571] [Q&A]
I have a use case where I am caching user's queries that goes to OpenAI. Now, for example, 2 different users can ask the same question, so the answer will be provided from the cache. But my need is to separate the cache based on user IDs as I need to pass some other context with minor changes. Is there any way to pass some unique ID and every time there is a hit from the cache it should be from that particular unique ID? The cache of 2 users must not mix up even if they ask the same question.

**Comment by SimFG** [comment]
You can use different cache object for these users.

**Reply by Amritpal2001** [reply]
Thanks for the reply.
Currently, I am keeping all cache questions/answers in SQLite. 
How can I filter that out for different users from a single SQLite database? It would be good if you could provide some example/docs link.

PS: Sorry for the late response.

**Reply by SimFG** [reply]
you can try to use the different table for the different cache.
link: https://gptcache.readthedocs.io/en/latest/references/manager.html#module-gptcache.manager.scalar_data.sql_storage

**Reply by Amritpal2001** [reply]
Thanks, @SimFG, this sounds good. However there will be many users, so creating a separate table for each user (different cache) is not feasible. Is there some way I can pass user id (unique id) everytime a new row gets created in the table and use that id to filter cache for that particular user?
